### PR TITLE
Cache worker tests

### PR DESCRIPTION
**Pull request by Mrowqa**
Depends on https://github.com/CraneStation/wasmtime/pull/337.

### CODE REVIEW

**Comment by Mrowqa** [comment]
TODO:
- [ ] the tests sometimes fail - fix it

One thing is they are not deterministic and depend on system clock (mtime when writing to the disk), so that's one possible reason - it could be the reason if setting mtime by the system is nondeterministic because of some optimizations.

File: wasmtime-environ/src/cache/worker.rs [hunk 1]
```diff
@@ -422,13 +438,29 @@ impl WorkerThread {
         trace!("Trying to clean up cache");
 
         let mut cache_index = self.list_cache_contents();
+        let future_tolerance = SystemTime::now()
+            .checked_add(
+                self.cache_config
+                    .allowed_clock_drift_for_files_from_future(),
+            )
+            .expect("Brace your cache, the end of the world is coming (time overflow)");
```
Comment (Mrowqa) [inline wasmtime-environ/src/cache/worker.rs]: I could actually call it "the next Big Bang" 😆

File: wasmtime-environ/src/cache/worker.rs [hunk 2]
```diff
@@ -833,9 +873,8 @@ fn is_fs_lock_expired(
     allowed_future_drift: Duration,
 ) -> bool {
     let mtime = match entry
-        .map(|e| e.metadata())
-        .unwrap_or_else(|| path.metadata())
-        .and_then(|metadata| metadata.modified())
+        .map_or_else(|| path.metadata(), |e| e.metadata())
+        .and_then(|metadata| metadata.modified()) // todo: research if we can change it to creation time (for the tests), if yes - it would be a little better to use it
```
Comment (sunfishcode) [inline wasmtime-environ/src/cache/worker.rs]: As we talked about offline, we can remove this todo and just use the `modified()` time. If there is somehow a process continually appending to the cache, it'll eventually consume all of disk space and get noticed :-).

Comment (Mrowqa) [reply wasmtime-environ/src/cache/worker.rs]: Done.

**Pull request merged**